### PR TITLE
cat update route is done

### PR DESCRIPTION
**Pull request by Maksadbek**

### CODE REVIEW

File: routes/cat.go [hunk 1]
```diff
@@ -42,7 +42,23 @@ func createCat(c web.C, w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		panic(err)
+	}
+
+	err = json.NewDecoder(r.Body).Decode(&cat)
+	panicOnErr(err)
+	cat.ID = id
+	err = cat.Update()
+	panicOnErr(err)
+	orderJSON, _ := json.Marshal(&cat)
```
Comment (mmirolim) [inline routes/cat.go]: why you are ignoring err?

File: routes/cat.go [hunk 2]
```diff
@@ -42,7 +42,23 @@ func createCat(c web.C, w http.ResponseWriter, r *http.Request) {
 }
 
 func updateCat(c web.C, w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "Update user %s", c.URLParams["id"])
+	id, err := strconv.Atoi(c.URLParams["id"])
+	panicOnErr(err)
+	var oldCat, cat ds.Cat
+	err = oldCat.FindOne(ds.Where{"id", "=", id})
+	panicOnErr(err)
+	if err != nil {
```
Comment (mmirolim) [inline routes/cat.go]: if redundant here

File: routes/cat.go (hunk 1 above)
Comment (Maksadbek) [inline routes/cat.go]:
```Go
func panicOnErr(err error) {
	if err != nil {
		panic(err)
	}
}
```

File: routes/cat.go [hunk 3]
```diff
@@ -42,7 +44,24 @@ func createCat(c web.C, w http.ResponseWriter, r *http.Request) {
...
+	panicOnErr(err)
+
+	err = json.NewDecoder(r.Body).Decode(&cat)
+	panicOnErr(err)
+
+	cat.ID = id
+	err = cat.Update()
+	panicOnErr(err)
+
+	orderJSON, _ := json.Marshal(&cat)
```
Comment (mmirolim) [inline routes/cat.go]: you are still ignoring err from json marshaling should be
```Go
orderJSON, err := json.Marshal(&cat)
panicOnErr(err)
```

File: routes/cat.go [hunk 4]
```diff
@@ -15,20 +15,39 @@ func getCatAll(c web.C, w http.ResponseWriter, r *http.Request) {
 	var err error
 	sts, err := st.FindAll(ds.Where{"id", ">", 0}, 0)
 	panicOnErr(err)
-	jsn, err := json.Marshal(sts)
-	fmt.Fprintf(w, string(jsn))
+
+	catsJSON, err := json.Marshal(sts)
+	if err != nil {
```
Comment (mmirolim) [inline routes/cat.go]: Don't forget about DRY principle
```go
if err != nil {
		res := "{'result': 'failure'}"
		result, err := json.Marshal(res)
		fmt.Fprintf(w, string(result))
		panicOnErr(err)
	} else {
		fmt.Fprintf(w, string(catsJSON))
	}
```
same code you wrote many times, how about to create function to encapsulate this functionality

Comment (mmirolim) [inline routes/cat.go]: Something like
```go
func sendResponse(err, w http.ResponseWriter, json []byte) error {
     opFail := OperationStatus{"fail"}    
    if err != nil {
        r, err := json.Marshal(opFail)
        if err != nil {
              return err
        }
        fmt.Fprintf(w, string(r))
 } else {
        fmt.Fprintf(w, string(json))
}
type OperationStatus struct {
        Status string `json:"status"`
}
````
code is just example, but I hope you understand want I mean

Comment (Maksadbek) [inline routes/cat.go]: okay

**Pull request closed without merging**